src/metabase/query_processor/middleware/annotate.clj

```diff
@@ -517,10 +517,14 @@
 (defn- flow-field-metadata
   "Merge information about fields from `source-metadata` into the returned `cols`."
   [source-metadata cols dataset?]
-  (let [index           (fn [col] (or (:id col) (:name col "")))
-        index->metadata (m/index-by index source-metadata)]
-    (for [col cols]
-      (if-let [source-metadata-for-field (-> col index index->metadata)]
+  (let [by-key (m/index-by (comp qp.util/field-ref->key :field_ref) source-metadata)]
+    (for [{:keys [field_ref source] :as col} cols]
+     ;; aggregation fields are not from the source-metadata and their field_ref
+     ;; are not unique for a nested query. So do not merge them otherwise the metadata will be messed up.
+     ;; TODO: I think the best option here is to introduce a parent_field_ref so that
+     ;; we could merge preserve metadata such as “:sematic_type or :unit” from the source field.
```

Comment:
Not sure how I feel about the smart quote characters in code files TBH. I guess they're ok in comments. But I'm assuming something did this for you automatically... maybe you should disable that.